packages/interactivity-router/src/index.js

```diff
@@ -117,21 +126,45 @@ export const { state, actions } = store( 'core/router', {
 				setTimeout( resolve, options.timeout ?? 10000 )
 			);
 
+			// Don't update the navigation status immediately, wait 400 ms.
+			const timeout = setTimeout( () => {
+				if ( navigatingTo === href ) {
+					navigation.hasStarted = true;
+					navigation.hasFinished = false;
+					navigation.message = navigation.texts.loading;
+				}
+			}, 400 );
+
 			const page = yield Promise.race( [
 				pages.get( url ),
 				timeoutPromise,
 			] );
 
+			// Dismiss loading message if it hasn't been added yet.
+			clearTimeout( timeout );
+
 			// Once the page is fetched, the destination URL could have changed
 			// (e.g., by clicking another link in the meantime). If so, bail
 			// out, and let the newer execution to update the HTML.
 			if ( navigatingTo !== href ) return;
 
+			navigation.hasStarted = false;
+			navigation.hasFinished = true;
+			// Announce that the page has been loaded. If the message is the
+			// same, we use a no-break space similar to the @wordpress/a11y
+			// package: https://github.com/WordPress/gutenberg/blob/c395242b8e6ee20f8b06c199e4fc2920d7018af1/packages/a11y/src/filter-message.js#L20-L26
+			navigation.message =
+				navigation.texts.loaded +
+				( navigation.message === navigation.texts.loaded
+					? '\u00A0'
+					: '' );
+
 			if ( page ) {
 				renderRegions( page );
 				window.history[
 					options.replace ? 'replaceState' : 'pushState'
 				]( {}, '', href );
+				state.url = href;
 			} else {
 				window.location.assign( href );
 				yield new Promise( () => {} );
```

Comment:
Why was this here? 😆

It'd be great to remember the reason and add a comment so people don't delete it in the future.

Reply:
I added a comment in https://github.com/WordPress/gutenberg/pull/58377/commits/1c4445fdc4c0069571aa866c1a4797549fe7cb48.

Reply:
Oh, that was it! Thanks.